### ISSUE TITLE
Resolvers broken when having a unique, nullable field on the model

### ISSUE THREAD

**Issue by nickluger** [graphql-nexus/nexus-plugin-prisma#324]
I have this simple Prisma datamodel:

```graphql
type User {
  email: String @unique  // this field can be null!
  id: ID! @id
  permissions: [Permission]
}

type Permission {
  id: ID! @id
  user: User!
  name: String!
}
```

Now, for fetching a user and resolving the `permissions`-field, i use `nexus-prisma` like this:
```ts
export const User = prismaObjectType({
  name: "User",
  definition(t) {
    t.prismaFields([
      "email",
      "id",
      { name: "permissions", args: [] }
    ]);
```

If  a user is returned from the database, and **the user has the `email` field set**, the mutation **works** and the permissions are resolved correctly.

If a user **without `email` field** is returned though, resolving the `permissions`-field fails:
```
 You provided an invalid argument for the where selector on User ...
"path":["authenticateOrCreateUser","user","permissions"],"...
{"authenticateOrCreateUser":{"user":{"id":"cjxps3vcn3jbx087502rb73ei","name":"Taya Bahringer","email":null,"permissions":null},"isNewSignup":true}}
```

It seems to me, that `nexus-prisma` is trying to resolve the user's relationships (like `permissions`) using the unique but nullable field `email` instead of the `id` field. That being `null` leads to the exception above.

I have a gut feeling that it has something to do with this fix here:
https://github.com/prisma/nexus-prisma/issues/65

**Basically, if there's an `id`-field on the parent it should be preferred for all resolvers.**

Using the `id`-field to resolve the thing manually works, by the way:

```ts
  t.list.field("permissions", {
      type: "Permission",
      resolve: ({ id }, args, ctx) => ctx.db.user({ id }).permissions()
    });
```

**Comment by nickluger** [comment]
I don't know whether that's the reason, but if i change the _order of the fields_ in my `datamodel.prisma` and delete + redeploy the Prisma service & re-generate all the stuff the error goes away....

```
// Broken
 email: String @unique
 id: ID! @id
```

```
// Works 😅
 id: ID! @id
 email: String @unique
```

Do you kind of sequentially look for unique fields, and pick the first one you find?
